app/src/main/java/com/example/teamyuml/decentworkmobile/fragments/ListViewFragment.java

```diff
@@ -0,0 +1,210 @@
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.example.teamyuml.decentworkmobile.VolleyInstance;
+import com.example.teamyuml.decentworkmobile.views.CustomListView;
+import com.example.teamyuml.decentworkmobile.views.CustomWorkerView;
+import com.example.teamyuml.decentworkmobile.views.NoticeDetails;
+import com.example.teamyuml.decentworkmobile.views.WorkerDetails;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.Console;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class ListViewFragment extends Fragment {
+    private ArrayList<HashMap<String, String>> data;
+    private ListView listView;
+    private String URL;
+    private int listViewId;
+    private int listLayoutId;
+    private Method callMethod;
+    private String initClass;
+    private String packageName;
+    private ArrayAdapter<HashMap<String, String>> adapterClass;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle bundle) {
+        Bundle fields = getArguments();
+        data = new ArrayList<>();
+
+        try {
+            if (fields != null) {
+                initializeFragment(fields);
+                callMethod.invoke(this);
+            }
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            Toast.makeText(
+                getActivity() ,"Brak dostępu do metody.", Toast.LENGTH_LONG).show();
+            e.printStackTrace();
+        }
+
+        View v = inflater.inflate(listLayoutId, container, false);
+        listView = v.findViewById(listViewId);
+        return v;
+    }
+
+    /**
+     * Initialize fragment fields.
+     * @param bundle - Bundle from {@link Fragment#getArguments()} method
+     */
+    private void initializeFragment(Bundle bundle) throws NoSuchMethodException {
+        URL = bundle.getString("url");
+        listViewId = bundle.getInt("listViewId");
+        listLayoutId = bundle.getInt("listLayoutId");
+        String methodName = bundle.getString("methodName");
+        initClass = bundle.getString("initClass");
+        packageName = bundle.getString("packageName");
```

Comment:
Why We can't make it as class field? I mean package name is same for every view so We don't need to send it in Bundle IMO.